src/store.c

```diff
@@ -290,47 +357,48 @@ static bool writer_open(
     return false;
 }
 
-static struct encoder writer_begin(
-        struct rill_store *store,
-        const struct vals *vals,
-        struct indexer *indexer)
+static void writer_flush_indices(
+    struct rill_store *store,
+    struct indexer *indexer_a,
+    struct indexer *indexer_b)
 {
-    size_t len = sizeof(*vals) + sizeof(vals->data[0]) * vals->len;
-    assert(store->head->vals_off + len < store->vma_len);
+    const size_t indexer_a_size = indexer_cap(indexer_a->len);
+    const size_t indexer_b_size = indexer_cap(indexer_b->len);
 
-    memcpy(store->vals, vals, len);
+    // TODO: this should replicate what used to be done for indices
```

Comment:
FYI I believe this was here, talking about the assert that I might have removed to oblivion...